FabNESO.py

```diff
@@ -31,17 +37,38 @@ def neso(
         solver: Which NESO solver to use.
         conditions_file_name: Name of conditions XML file in configuration directory.
         mesh_file_name: Name of mesh XML in configuration directory.
+        **kwargs: Additional kwargs will be passed to ensemble_tools.edit_parameters to
+                  create a temporary conditions file with these overriden parameters
     """
-    fab.with_config(config)
-    fab.execute(fab.put_configs, config)
-    fab.job(
-        {
-            "script": "neso",
-            "neso_solver": solver,
-            "neso_conditions_file": conditions_file_name,
-            "neso_mesh_file": mesh_file_name,
-        }
+    # Use a temporary directory context so that we can handle parsameter inputs
+    # from the command line
+    original_config_path = Path(fab.find_config_file_path(config))
+    temporary_context: Union[TemporaryDirectory, nullcontext] = (
+        TemporaryDirectory(prefix=f"{config}_", dir=original_config_path.parent)
+        if not kwargs == {}
+        else nullcontext()
     )
```

Comment:
I had to add this little explicit pre-definition of the context to satisfy mypy. It was complaining during the with statement that it couldn't find the __enter__ and __exit__ consistently when there were differing types during the definition.